.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,19 +1,24 @@
-Add compact, short information about your PR for easier understanding:
+<!-- Add compact, short information about your PR for easier understanding, like the objective of the PR -->
 
-- Goal of the PR
-- How does the PR work?
-- Does it resolve any reported issue?
-- If not a bug fix, why is this PR needed? What usecases does it solve?
+### How does the PR work?
```

Comment:
I dont feel like the heading here is really needed

Reply:
This heading encourages PR authors to elaborate upon implementation details. The comment below this line indicates that this heading is optional:
```markdown
<!-- Elaborate on the changes, if applicable, or delete this heading -->
```

Are you sure this heading should be removed?

Reply:
i'm not <i>sure</i>, it's just personal preference
that said, the author can still be encouraged to elaborate even if the actual heading is gone

Reply:
Makes sense. I'll remove it.

Reply:
Most PRs are non-trivial, and elaborating on their implementation, which would at least serve as documentation, should be encouraged by default, IMO. And ofc, authors can always delete the heading if it doesn't make sense for their PR. Again, that's just my opinion, feel free to let me know if this should be removed.